src/main/java/com/aws/greengrass/device/ClientDevicesAuthService.java

```diff
@@ -99,9 +98,31 @@ public ClientDevicesAuthService(Topics topics, GroupManager groupManager, Certif
     protected void install() throws InterruptedException {
         super.install();
         //handleConfiguration
-        this.deviceGroupsTopics = this.config.lookupTopics(CONFIGURATION_CONFIG_KEY, DEVICE_GROUPS_TOPICS);
-        this.deviceGroupsTopics.subscribe(this::handleConfigurationChange);
-        this.config.lookup(CONFIGURATION_CONFIG_KEY, CA_TYPE_TOPIC).subscribe(this::updateCAType);
+        this.config.lookupTopics(CONFIGURATION_CONFIG_KEY).subscribe((whatHappened, node) -> {
+            if (whatHappened == WhatHappened.timestampUpdated || whatHappened == WhatHappened.interiorAdded) {
+                return;
+            }
+            logger.atTrace().kv("why", whatHappened).kv("node", node).log();
+            Topics deviceGroupTopics = this.config.lookupTopics(CONFIGURATION_CONFIG_KEY, DEVICE_GROUPS_TOPICS);
+            Topic caTypeTopic = this.config.lookup(CONFIGURATION_CONFIG_KEY, CA_TYPE_TOPIC);
+
+            if (whatHappened == WhatHappened.initialized) {
+                handleConfigurationChange(whatHappened, deviceGroupTopics);
+                updateCAType(whatHappened, caTypeTopic);
+            } else if (DEVICE_GROUPS_TOPICS.equals(node.getName())) {
+                handleConfigurationChange(whatHappened, deviceGroupTopics);
+            } else if (CA_TYPE_TOPIC.equals(node.getName())) {
+                // Skip the first duplicate event of childChanged that is received after initialize.
+                // This event is same as the one received in initialize. Skip doing duplicate work.
+                // Parse any further childChanged events.
+                if (!hasChildChangedAfterInitialize && whatHappened == WhatHappened.childChanged) {
+                    hasChildChangedAfterInitialize = true;
+                    return;
+                }
+                updateCAType(whatHappened, caTypeTopic);
```

Comment:
This doesn't feel right - is this a workaround to current Nucleus behavior or is this part of a contract? In either case, `updateCAType` should be idempotent so there shouldn't be a problem calling this twice. Unless the CA type changes it will just reload the existing CA.

Reply:
Ahhh I over complicated it. 
It was a workaround.
It was just a matter of checking if the `caTypeTopic` is null. Updated it in the new commit.